app/src/utils/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+function replaceItemAtIndex<T>(arr: T[], index: number, newValue: T) {
+  return [...arr.slice(0, index), newValue, ...arr.slice(index + 1)];
+}
+
+function removeItemAtIndex<T>(arr: T[], index: number) {
+  return [...arr.slice(0, index), ...arr.slice(index + 1)];
+}
+
+function findArrayIndex<T>(arr: T[], predicate: (item: T) => boolean) {
+  return arr.findIndex(predicate);
+}
+
+function findNumberOfPropertyValuesInArrayOfObjects<T>(
+  objArr: T[],
+  property: keyof T
+): React.ReactNode {
+  const categoriesInobjArr: Record<string, number> = {};
+  objArr.forEach((obj) => {
+    const qty = (obj as { quantity: number }).quantity || 0;
+    categoriesInobjArr[obj[property] as string] =
+      (categoriesInobjArr[obj[property] as string] + qty || 0) + 1;
+  });
+  return Object.entries(categoriesInobjArr).map((el) => {
+    return (
+      <>
+        <strong className="text-lg">{el[1]}</strong>
+        {' x '}
+        {el[0]}
+        {'  '}
+      </>
+    );
+  });
+}
```

Comment:
this seems to be quite a specific function. It starts by taking a generic array but then you use a cast which is no longer generic. `const qty = (obj as { quantity: number }).quantity || 0;`
I'll have a think about this for a bit and come back to you

Reply:
Yeah I tried to make it as reusable as possible. It can be used to count the number of times different values of the same potentially defined property appear in different objects of the same array. Up to here I think it is a general purpose funct. I included the extra bit of logic of adding to the total count of each property value, the number of a potentially defined property qty. And yes this bit makes it non-general but if the prop does not exists nothing happens. But still not the most appropriate solution???